src/main/scala/utility/Sort.scala

```diff
@@ -0,0 +1,151 @@
+/***************************************************************************************
+* Copyright (c) 2024 Beijing Institute of Open xVec Chip (BOSC)
+* Copyright (c) 2024 Institute of Computing Technology, Chinese Academy of Sciences
+*
+* XiangShan is licensed under Mulan PSL v2.
+* You can use this software according to the terms and conditions of the Mulan PSL v2.
+* You may obtain a copy of Mulan PSL v2 at:
+*          http://license.coscl.org.cn/MulanPSL2
+*
+* THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
+* EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
+* MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+*
+* See the Mulan PSL v2 for more details.
+* 
+* Acknowledgement
+* This implementation is inspired by several key materials:
+* [1] https://github.com/OpenXiangShan/XiangShan/blob/d4ca2db4b3d45add495358a040aa2b9f26e40370/src/main/scala/xiangshan/mem/prefetch/L1PrefetchComponent.scala#L103
+* [2] https://github.com/freechipsproject/chisel-bootcamp/blob/master/2.3_control_flow.ipynb
+***************************************************************************************/
+
+package utility
+
+import chisel3._
+import chisel3.util._
+
+class TwithPtr[A <: Data, B <: CircularQueuePtr[B]](a: A, b: B) extends Bundle {
```

Comment:
What is Twith? Should we use a better name?

Reply:
T is an abbreviation for "template", which is a `Data` template with a `Ptr`.

Reply:
I see. Then DataWithPtr may be a clearer name.

Reply:
I see. Then `DataWithPtr` may be a better name.